lib/rex/exploitation/cmdstager/echo.rb

```diff
@@ -26,10 +26,14 @@ def initialize(exe)
   # and initialize opts[:enc_format].
   #
   def generate(opts = {})
-    opts[:temp] = opts[:temp] || '/tmp/'
-    opts[:temp].gsub!(/\\/, "/")
-    opts[:temp] = opts[:temp].shellescape
-    opts[:temp] << '/' if opts[:temp][-1,1] != '/'
+    if opts[:temp] == false
```

Comment:
Allowing options to be both strings and parameters isn't a good idea. Makes even harder to remind the format of the available options. And it's already hard to remember :)

I'll fix before landing.